packages/maskbook/src/web3/hooks/useTokenTransferCallback.ts

```diff
@@ -7,21 +7,31 @@ import { useEtherTransferCallback } from './useEtherTransferCallback'
 export function useTokenTransferCallback(
     type: EthereumTokenType,
     address: string,
-    amount: string,
+    amountOrTokenId: string,
     recipient: string,
-    memo: string,
+    memo?: string,
 ) {
-    const r1 = useEtherTransferCallback(type === EthereumTokenType.Ether ? amount : '', recipient, memo)
-    const r2 = useERC20TokenTransferCallback(address, type === EthereumTokenType.ERC20 ? amount : '', recipient)
-    const r3 = useERC721TokenTransferCallback(address, type === EthereumTokenType.ERC721 ? amount : '', recipient)
+    const r1 = useEtherTransferCallback(type === EthereumTokenType.Ether ? amountOrTokenId : '', recipient, memo)
+    const r2 = useERC20TokenTransferCallback(
+        address,
+        type === EthereumTokenType.ERC20 ? amountOrTokenId : '',
+        recipient,
+    )
+    const r3 = useERC721TokenTransferCallback(
+        address,
+        type === EthereumTokenType.ERC721 ? amountOrTokenId : '',
+        recipient,
+    )
     const type_ = type
     switch (type_) {
         case EthereumTokenType.Ether:
             return r1
         case EthereumTokenType.ERC20:
             return r2
         case EthereumTokenType.ERC721:
-            throw new Error('Not implemented')
+            return r3
+        case EthereumTokenType.ERC1155:
+            throw new Error('To be implemented.')
```

Comment:
what about those To be done?

Reply:
The token transfer logic upon ERC1155 contract.

Reply:
We are going to support ERC1155 in the future.